kinto/views/permissions.py

```diff
@@ -125,6 +133,11 @@ def get_records(self, filters=None, sorting=None, pagination_rules=None,
                 # (e.g "bucket:write" gives "bucket:read" but not "group:read")
                 permissions |= obtained[resource_name]
 
+                # For consistency with event payloads, change top level bucket
+                # resource_name from '' to bucket
+                if resource_name == '':
+                    resource_name = 'bucket'
```

Comment:
this would not be necessary if `view_lookup()` would return the right value

Reply:
From my comment on `kinto/core/utils.py`, I need `''` to get the permissions in:

https://github.com/Kinto/kinto/blob/6d77dcb2b94f9198fa08db489776440f7df0c275/kinto/views/permissions.py#L123

Then I have to switch it back to `bucket` for the response payload the user sees. If I didn't make this change, the response would be:

```
"data": [
        {
            "permissions": [
                "bucket:create"
            ],
            "resource_name": "",
            "uri": "/buckets"
        }
    ]
```

Rather than:

```
"data": [
        {
            "permissions": [
                "bucket:create"
            ],
            "resource_name": "bucket",
            "uri": "/buckets"
        }
    ]
```